src/de/deserialize.rs

```diff
@@ -312,17 +309,17 @@ mod test {
 
     #[test]
     fn deserialize_good_publish() {
-        let mut serialized_publish: [u8; 7] = [
+        let serialized_publish: [u8; 7] = [
             0x30, // Publish, no QoS
-            0x04, // Remaining length
```

Comment:
I'm somewhat unsure why this was passing in the past. It looks completely wrong to me - there's obviously 5 bytes following the packet.